src/sdk/namespace_merge.js

```diff
@@ -25,9 +25,18 @@ class NamespaceMerge {
     // OBJECT LIST //
     /////////////////
 
-    list_objects(params, object_sdk) {
-        return this._ns_map(ns => ns.list_objects(params, object_sdk))
-            .then(res => this._handle_list(res, params));
+    async list_objects(params, object_sdk) {
+        let res = (await this._ns_map(async function(ns) {
+            try {
+                let ns_objects = await ns.list_objects(params, object_sdk);
+                return ns_objects;
+            } catch (err) {
+                if (err.code === 'NoSuchBucket') return;
+                throw err;
+            }
+        })).filter(objects => objects);
+        if (!res || !res.length) throw new S3Error(S3Error.NoSuchBucket);
```

Comment:
Once again the `_throw_if_any_failed_or_get_succeeded` method in namespace_merge handles that case scenario if you call the ns_map with `EXCEPT_REASONS` as I've stated above.